alchemy/src/cloudflare/bucket.ts

```diff
@@ -202,88 +225,119 @@ export function isBucket(resource: Resource): resource is R2Bucket {
  *
  * @see https://developers.cloudflare.com/r2/buckets/
  */
-export const R2Bucket = Resource(
+export async function R2Bucket(
+  id: string,
+  props: BucketProps = {},
+): Promise<R2Bucket> {
+  return await _R2Bucket(id, {
+    ...props,
+    dev: {
+      ...(props.dev ?? {}),
+      force: Scope.current.local,
+    },
+  });
+}
+
+const _R2Bucket = Resource(
   "cloudflare::R2Bucket",
   async function (
     this: Context<R2Bucket>,
-    _id: string,
+    id: string,
     props: BucketProps = {},
   ): Promise<R2Bucket> {
-    const api = await createCloudflareApi(props);
-    const bucketName = props.name || this.id;
+    const bucketName = props.name || id;
     const allowPublicAccess = props.allowPublicAccess === true;
+    const dev = {
+      id: this.output?.dev?.id ?? bucketName,
+      remote: props.dev?.remote ?? false,
+    };
 
-    switch (this.phase) {
-      case "create": {
-        const bucket = await createBucket(api, bucketName, props).catch(
-          async (err) => {
-            if (
-              err instanceof CloudflareApiError &&
-              err.status === 409 &&
-              props.adopt
-            ) {
-              return await getBucket(api, bucketName, props);
-            }
-            throw err;
-          },
+    if (this.scope.local && !props.dev?.remote) {
+      return this({
```

Comment:
Yeah, this is why I don't liek switch and never use it. Not expressive enough and awkward syntax.